# Assignment 1
### Understanding Uncertainty
### Due 9/5

1. Create a new public repo on Github under your account. Include a readme file.
2. Clone it to your machine. Put this file into that repo.
3. Use the following function to download the example data for the course:

In [14]:
def download_data(force=False):
    """Download and extract course data from Zenodo."""
    import urllib.request, zipfile, os
    
    zip_path = 'data.zip'
    data_dir = 'data'
    
    if not os.path.exists(zip_path) or force:
        print("Downloading course data")
        urllib.request.urlretrieve(
            'https://zenodo.org/records/16954427/files/data.zip?download=1',
            zip_path
        )
        print("Download complete")
    else:
        print("Download file already exists")
        
    if not os.path.exists(data_dir) or force:
        print("Extracting data files...")
        with zipfile.ZipFile(zip_path, 'r') as zip_ref:
            zip_ref.extractall(data_dir)
        print("Data extracted")
    else:
        print("Data directory already exists")

download_data()

Download file already exists
Data directory already exists


4. Open one of the datasets using Pandas:
    1. `ames_prices.csv`: Housing characteristics and prices
    2. `college_completion.csv`: Public, nonprofit, and for-profit educational institutions, graduation rates, and financial aid
    3. `ForeignGifts_edu.csv`: Monetary and in-kind transfers from foreign entities to U.S. educational institutions
    4. `iowa.csv`: Liquor sales in Iowa, at the transaction level
    5. `metabric.csv`: Cancer patient and outcome data
    6. `mn_police_use_of_force.csv`: Records of physical altercations between Minnessota police and private citizens
    7. `nhanes_data_17_18.csv`: National Health and Nutrition Examination Survey
    8. `tuna.csv`: Yellowfin Tuna Genome (I don't recommend this one; it's just a sequence of G, C, A, T )
    9. `va_procurement.csv`: Public spending by the state of Virginia

In [15]:
import pandas as pd
import numpy as np

In [16]:
nhanes = pd.read_csv('/Users/khansaamaa/Documents/GitHub/DS-5030-Understanding-Uncertainty/data/nhanes_data_17_18.csv')

/var/folders/_9/mppjh83s7vv1krgq3bp31dnr0000gn/T/ipykernel_24020/3691063809.py:1: DtypeWarning: Columns (142) have mixed types. Specify dtype option on import or set low_memory=False.
  nhanes = pd.read_csv('/Users/khansaamaa/Documents/GitHub/DS-5030-Understanding-Uncertainty/data/nhanes_data_17_18.csv')


In [17]:
nhanes.head()
nhanes.info()
nhanes.describe()
nhanes.shape
nhanes.dtypes
nhanes.columns

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8366 entries, 0 to 8365
Columns: 198 entries, SEQN to TimesLost10LbsOrMoreToLoseWeight
dtypes: float64(162), object(36)
memory usage: 12.6+ MB


Index(['SEQN', 'GeneralHealthCondition', 'EverBreastfedOrFedBreastmilk',
       'AgeStoppedBreastfeedingdays', 'AgeFirstFedFormuladays',
       'AgeStoppedReceivingFormuladays', 'AgeStartedOtherFoodbeverage',
       'AgeFirstFedMilkdays', 'TypeOfMilkFirstFedWholeMilk',
       'TypeOfMilkFirstFed2Milk',
       ...
       'DaysSmokedCigsDuringPast30Days', 'AvgCigarettesdayDuringPast30Days',
       'TriedToQuitSmoking', 'TimesStoppedSmokingCigarettes',
       'HowLongWereYouAbleToStopSmoking',
       'UnitOfMeasureDayweekmonthyear_2_SMQ',
       'CurrentSelfreportedHeightInches', 'CurrentSelfreportedWeightPounds',
       'TriedToLoseWeightInPastYear', 'TimesLost10LbsOrMoreToLoseWeight'],
      dtype='object', length=198)

In [22]:
nhanes.dtypes

SEQN                                   float64
GeneralHealthCondition                  object
EverBreastfedOrFedBreastmilk           float64
AgeStoppedBreastfeedingdays            float64
AgeFirstFedFormuladays                 float64
                                        ...   
UnitOfMeasureDayweekmonthyear_2_SMQ     object
CurrentSelfreportedHeightInches        float64
CurrentSelfreportedWeightPounds        float64
TriedToLoseWeightInPastYear            float64
TimesLost10LbsOrMoreToLoseWeight        object
Length: 198, dtype: object

5. Pick two or three variables and briefly analyze them
    - Is it a categorical or numeric variable?
    - How many missing values are there? (`df['var'].isna()` and `np.sum()`)
    - If categorical, tabulate the values (`df['var'].value_counts()`) and if numeric, get a summary (`df['var'].describe()`)

Variable 1 :

Name : GeneralHealthCondition
Datatype : object
Variabletype : categorical
Missing Values : 2402
Total Values : 8366

In [18]:
nhanes['GeneralHealthCondition'].value_counts()

GeneralHealthCondition
Good         2454
Very good    1544
Fair or      1175
Excellent     619
Poor?         172
Name: count, dtype: int64

In [12]:
nhanes['GeneralHealthCondition'].isna()
nhanes['GeneralHealthCondition'].isna().sum()
#    - If categorical, tabulate the values (`df['var'].value_counts()`) and if numeric, get a summary (`df['var'].describe()`)

2402

Variable 2 :

Name : AgeStartedOtherFoodbeverage
Datatype : float64
Variabletype : numeric
Missing Values : 7289
Total Values : 8366

In [20]:
nhanes['AgeStartedOtherFoodbeverage'].describe()

count      1077.000000
mean      16604.331476
std      121232.078614
min           1.000000
25%         121.000000
50%         182.000000
75%         243.000000
max      999999.000000
Name: AgeStartedOtherFoodbeverage, dtype: float64

In [21]:
nhanes['AgeStartedOtherFoodbeverage'].isna()
nhanes['AgeStartedOtherFoodbeverage'].isna().sum()

7289

6. What are some questions and prediction tools you could create using these data? Who would the stakeholder be for that prediction tool? What practical or ethical questions would it create? What other data would you want, that are not available in your data?

1. Possible Questions & Prediction Tools


- Predictive Health Risk Models

    - Can we predict whether a person is at risk of developing diabetes, hypertension, or heart disease based on BMI, blood pressure, cholesterol, and lifestyle factors (smoking, exercise, diet)?

- Quality of Life Forecasting

    - Can we predict a person’s self-reported general health condition based on clinical measures and lifestyle indicators?

- Healthcare Access & Disparities

    - How do socioeconomic variables (income, insurance, education) predict access to preventive care?

- Nutrition & Obesity Risk

    - How do dietary recall data predict obesity risk or metabolic syndrome?

2. Stakeholders for Prediction Tools

- Public health agencies (CDC, NIH) → to design better interventions.

- Clinicians & hospitals → to identify at-risk patients early.

- Policy makers → to address health inequities.

- Insurance companies (with caution) → to assess population-level risk, though this raises ethical issues.

- Researchers → to explore relationships between lifestyle and health outcomes.

3. Practical & Ethical Questions

- Bias and Fairness: Does the model unfairly disadvantage certain racial/ethnic or income groups?

- Privacy: How is sensitive health data protected?

- Consent: Were participants informed their data might be used for predictive modeling?

- Misuse: Could insurers/employers use these predictions to deny coverage or employment?

- Interpretability: Can predictions be explained clearly to non-experts?

4. Additional Data That Would Be Useful

- Longitudinal outcomes (follow-up health outcomes instead of one-time survey).

- Genetic data (some NHANES subsets have this, but not always).

- Environmental exposures (air pollution, neighborhood food access).

- Healthcare utilization (hospital visits, medications adherence).

- Behavioral tracking (physical activity from wearables, sleep data).

- Psychosocial measures (stress, social support).

7. Commit your work to the repo (`git commit -am 'Finish assignment'` at the command line, or use the Git panel in VS Code). Push your work back to Github and submit the link on Canvas in the assignment tab.